spyder/utils/programs.py

```diff
@@ -710,7 +712,27 @@ def run_python_script_in_terminal(fname, wdir, args, interact,
                 else:
                     run_program(program['cmd'], arglist)
                 return
-        # TODO: Add a fallback to OSX
+    elif sys.platform == 'darwin':
+        f = tempfile.NamedTemporaryFile('wt', suffix='.sh',
+                                        dir=get_temp_dir(), delete=False)
+        f.write(' '.join(p_args))
+        f.close()
+        os.chmod(f.name, 0o777)
+
+        def run_terminal_thread():
+            if wdir:
+                proc = subprocess.Popen(
+                    'open -a Terminal.app ' + f.name, cwd=wdir, shell=True)
+            else:
+                proc = subprocess.Popen(
+                    'open -a Terminal.app ' + f.name, shell=True)
+            # Prevent race condition
+            time.sleep(3)
+            proc.wait()
+            os.remove(f.name)
+
+        thread = threading.Thread(target=run_terminal_thread)

```

Comment:
Why not use QThread instead of threading here? Just asking, I don't know if it makes any difference or not.

Reply:
There's no difference between them in this case

Reply:
I guess it is better to use a QThread to keep things more in sync with the rest of the application?